### PR TITLE
feat: add hdwallet-metamask-shapeshift-multichain adapter

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
Disregard the below, this is now published as alpha: https://www.npmjs.com/package/@shapeshiftoss/metamask-snaps-adapter

Note, we had to bump tsc here to accomodate for the version in metamask-snaps monorepo (which we can't avoid because of new TS features from TS 4.4+ that need to be supported in some snaps transient dependencies), which subsequently broke a lot of things at type-level, explaining the disgusting amount of ts-ignores in this diff. 

~~Opening draft to visualize the diff
Note, both https://github.com/shapeshift/metamask-snaps/pull/100, https://github.com/shapeshift/hdwallet/pull/628, and https://github.com/shapeshift/web/pull/5166 assume verdaccio running on default port 4873, and all packages published there:~~

~~image~~

~~This is only for development purposes, and config should be reverted to default uplink before merging each of these PRs.~~

### CODE REVIEW

File: packages/hdwallet-metamask-shapeshift-multichain/src/binance.ts
Comment (gomesalexandre) [inline packages/hdwallet-metamask-shapeshift-multichain/src/binance.ts:1]: These do not conform to `ETHWallet` and means we'd have to do the discrimination in chain-adapters. Not saying that's the wrong approach but as I'm just starting to read this and only am one coffee in, just leaving this here for context

File: packages/hdwallet-native/src/crypto/isolation/adapters/binance.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ const bnbSdk = PLazy.from(() => import("bnb-javascript-sdk-nobroadcast"));
 
 export default {
   async create(keyPair: BIP32.Node): Promise<SigningDelegate> {
-    return async (tx: Transaction, signMsg?: any): Promise<Transaction> => {
+    return await (tx: Transaction, signMsg?: any): Promise<Transaction> => {
```
Comment (gomesalexandre) [inline packages/hdwallet-native/src/crypto/isolation/adapters/binance.ts:12]: ser

File: packages/hdwallet-metamask-shapeshift-multichain/src/binance.ts
Comment (gomesalexandre) [reply packages/hdwallet-metamask-shapeshift-multichain/src/binance.ts:1]: Hold my coffee, this is BNB wallet, not related to the BSC EVM chain, remove me

**Comment by gomesalexandre** [comment]
Opening because CI is now green, performing second pass self-review

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* master
  * **PR #628** <a href="https://app.graphite.dev/github/pr/shapeshift/hdwallet/628" target="_blank">Graphite</a>  👈
    * **PR #629** <a href="https://app.graphite.dev/github/pr/shapeshift/hdwallet/629" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/hdwallet/628?utm_source=stack-comment).

File: integration/package.json [hunk 2]
```diff
@@ -10,24 +10,24 @@
...
-    "@shapeshiftoss/hdwallet-portis": "1.50.4",
-    "@shapeshiftoss/hdwallet-trezor": "1.50.4",
-    "@shapeshiftoss/hdwallet-xdefi": "1.50.4",
+    "@shapeshiftoss/hdwallet-core": "^1.50.5-alpha.68",
+    "@shapeshiftoss/hdwallet-keepkey": "^1.50.5-alpha.68",
+    "@shapeshiftoss/hdwallet-keepkey-nodewebusb": "^1.50.5-alpha.68",
+    "@shapeshiftoss/hdwallet-keepkey-tcp": "^1.50.5-alpha.68",
+    "@shapeshiftoss/hdwallet-ledger": "^1.50.5-alpha.68",
+    "@shapeshiftoss/hdwallet-native": "^1.50.4",
+    "@shapeshiftoss/hdwallet-portis": "^1.50.4",
```
Comment (kaladinlight) [inline integration/package.json:19]: everything should be pinned to same version and regen lock file
```suggestion
    "@shapeshiftoss/hdwallet-native": "^1.50.4-alpha.68",
    "@shapeshiftoss/hdwallet-portis": "^1.50.4-alpha.68",
```

File: examples/sandbox/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hdwallet-sandbox",
-  "version": "1.50.4",
+  "version": "1.50.5-alpha.68",
```
Comment (kaladinlight) [inline examples/sandbox/package.json:3]: made the mistake of bumping package to non alpha in the metamask-snaps, but if we can get an update version whitelisted, would be nice to make this a non alpha release now that we are live I feel like.

File: integration/package.json (hunk 2 above)
Comment (kaladinlight) [reply integration/package.json:19]: looks like lerna is on `alpha.72` actually. all packages should be fixed to the same version if using lerna

Comment (kaladinlight) [reply integration/package.json:19]: going to push an update to change that and also bump `metamask-snaps-adapter` to the latest alpha version

**Review by gomesalexandre — APPROVED** [review]

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**